src/it/scala/SqsSpec.scala

```diff
@@ -30,6 +33,47 @@ class SqsSpec extends IntegrationSpec {
     }
   }
 
+  it should "keep consuming even after a consumer exception" in {
+    val msgEx = TestType("fail", -1)
+    val msg1 = TestType("ok1", 1)
+    val msg2 = TestType("ok2", 2)
+    val msg3 = TestType("ok3", 3)
+
+    producer.produce(msg1)
+      .flatMap(_ => producer.produce(msgEx))
+      .flatMap(_ => producer.produce(msg2))
+      .flatMap(_ => producer.produce(msg3)).map { _ =>
+      probe expectMsgAllOf (msg1, msg2, msg3)
+      succeed
+    }
+  }
+
+  it should "keep consuming even after a decode exception" in {
+    val msgEx = Promise[String]
+    val msg1 = TestType("ok1", 1)
+    val msg2 = TestType("ok2", 2)
+    val msg3 = TestType("ok3", 3)
+    
+    amazonSqsConn.sendMessageAsync("http://localhost:9324/queue/integration-test-q", "{bad formated message}",
+      new AsyncHandler[SendMessageRequest, SendMessageResult] {
+        override def onError(exception: Exception) = {
+          msgEx.failure(exception)
+        }
+        override def onSuccess(request: SendMessageRequest, result: SendMessageResult) = {
+          msgEx.success("ok")
+        }
+      })
```

Comment:
This part should be in Helper, that to both simplify the code and to make it more readable.